router/batchrouter/asyncdestinationmanager/salesforce-bulk/utils.go

```diff
@@ -0,0 +1,206 @@
+package salesforcebulk
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/sha256"
+	"encoding/csv"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/jsonrs"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+)
+
+const (
+	maxFileSize = 100 * 1024 * 1024 // 100MB in bytes
+)
+
+type ObjectInfo struct {
+	ObjectType      string
+	ExternalIDField string
+	ExternalIDValue string
+}
+
+func readJobsFromFile(filePath string) ([]common.AsyncJob, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("opening file: %w", err)
+	}
+	defer file.Close()
+
+	var jobs []common.AsyncJob
+	scanner := bufio.NewScanner(file)
+	scanner.Buffer(nil, 50000*1024)
+
+	for scanner.Scan() {
+		var job common.AsyncJob
```

Comment:
Instead of using a common job, can we use a custom map to avoid lookup for things like `jobID`, `externalId` or `rudderOperation`?

```suggestion
type AsyncJob struct {
	Message  map[string]interface{} `json:"message"`
	Metadata Metadata               `json:"metadata"`
}

type Metadata struct {
	JobID      int64  `json:"job_id"`
	Operation  string `json:"operation"`
	ExternalID string `json:"externalId"`
}
```